test/tests/userobjects/pre_aux_based_on_exec_flag/tests

```diff
@@ -0,0 +1,22 @@
+[Tests]
+  issues = '#18189'
+  design = '/GeneralUserObject.md'
```

Comment:
You will need to improve a markdown file to explain the dependency a little more. I remember we have a markdown file explaining how a uo can get into pre_aux group. You will need to find it.